### PR TITLE
placement: remove isolationlevel

### LINKED ISSUE #30858: placement: mirror issue of tikv/pd#4467
## Enhancement

Refer to the issue https://github.com/tikv/pd/issues/4467#issuecomment-995879181

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: close #30858 tikv/pd#4467

Problem Summary: `IsolationLevel` takes higher priority than rules of placement rules. So when it is set, placement rules may be violated, which is unwanted. So it is removed.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
Start tiup with 6 tikv, three of them with `region: r1`, others with `region: r2`.

Create a table with `CONSTRAINTS=[+region=r1] FOLLOWERS=2`, then `show placement`, you should see it is scheduled.

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/placement/rule.go [hunk 1]
```diff
@@ -58,7 +58,6 @@ func NewRule(role PeerRoleType, replicas uint64, cnst Constraints) *Rule {
 		Count:          int(replicas),
 		Constraints:    cnst,
 		LocationLabels: []string{"region", "zone", "rack", "host"},
-		IsolationLevel: "region",
```
Comment (lcwangchao) [inline ddl/placement/rule.go:61]: Need to modify some test？

Comment (xhebox) [reply ddl/placement/rule.go:61]: I did not find one by grep, i.e. I don't find any other assignment to this field. I remembered that tests will assert on rules only.

Comment (xhebox) [reply ddl/placement/rule.go:61]: Addressed. Remind me if you find another test needs to be fixed.

**Review by lcwangchao — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**